test/js.go

```diff
@@ -0,0 +1,73 @@
+//go:build js
```

Comment:
move this file to a test/js package

add readme / doc.go explaining how the testing of this stuffs is done.